mlflow/sklearn/utils.py

```diff
@@ -129,13 +138,24 @@ def _get_metrics_value_dict(metrics_list):
             _log_warning_for_metrics(metric.name, metric.function, e)
         else:
             metric_value_dict[metric.name] = metric_value
-
     return metric_value_dict
 
 
+def _get_artifacts_list(artifacts_list):
```

Comment:
@harupy I wonder if it makes sense to resolve these artifacts one-by-one within the function where we persist the artifacts. Otherwise, we create `n` matplotlib figures here and close them one-by-one as we log them. If we end up logging a larger volume of artifacts eventually / for certain use cases, or if some systems suffer degraded performance as a result of having 10-20 handles open, this could become problematic.

Reply:
Makes sense. I'll make the change.